--- /tmp/pth-2.0.7/pth.h.in	2006-06-08 13:54:02.000000000 -0400
+++ pth-2.0.7/pth.h.in	2012-01-13 18:31:10.000000000 -0500
@@ -43,7 +43,7 @@
 #include <time.h>          /* for struct timespec */
 #include <sys/time.h>      /* for struct timeval  */
 #include <sys/socket.h>    /* for sockaddr        */
-#include <sys/signal.h>    /* for sigset_t        */
+#include <signal.h>    /* for sigset_t        */
 @EXTRA_INCLUDE_SYS_SELECT_H@
 
     /* fallbacks for essential typedefs */
