diff --git a/npth/configure.ac b/npth/configure.ac
index c32480e..8d10564 100644
--- a/npth/configure.ac
+++ b/npth/configure.ac
@@ -215,7 +215,17 @@ if test "$have_w32_system" = no; then
   if test "$have_pthread" != no; then
     AC_DEFINE(HAVE_PTHREAD,1,[Define if we have pthread.])
     AC_CHECK_FUNCS([pthread_tryjoin_np pthread_setname_np pthread_getname_np])
-    AC_CHECK_FUNCS([pthread_mutex_timedlock])
+    # Android seems to include pthread_mutex_timedlock() in its headers, but
+    # it is not implemented.  Which means we can't really detect that.
+    # http://code.google.com/p/android/issues/detail?id=2138
+    case "${host}" in
+         *-*-android*)
+         echo "Android does not actually have pthread_mutex_timedlock()"
+         ;;
+         *)
+         AC_CHECK_FUNCS([pthread_mutex_timedlock])
+         ;;
+    esac
   fi
 fi
 
diff --git a/npth/src/npth.c b/npth/src/npth.c
index 07d228b..ab1bf50 100644
--- a/npth/src/npth.c
+++ b/npth/src/npth.c
@@ -297,7 +297,8 @@ npth_mutex_timedlock (npth_mutex_t *mutex, const struct timespec *abstime)
   return err;
 }
 
-
+/* Android does not have pthread_rwlock_* at all */
+#ifndef __ANDROID__
 int
 npth_rwlock_rdlock (npth_rwlock_t *rwlock)
 {
@@ -368,6 +369,7 @@ npth_rwlock_timedwrlock (npth_rwlock_t *rwlock, const struct timespec *abstime)
   LEAVE();
   return err;
 }
+#endif /* NOT __ANDROID__ */
 
 
 int
diff --git a/npth/src/npth.h.in b/npth/src/npth.h.in
index 9fdaa55..3ef2012 100644
--- a/npth/src/npth.h.in
+++ b/npth/src/npth.h.in
@@ -55,6 +55,7 @@
 #@INSERT_SYS_SELECT_H@
 
 #include <pthread.h>
+#include <sched.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -110,7 +111,7 @@ int npth_create(npth_t *thread, const npth_attr_t *attr,
 /* No can do! */
 /* pth_suspend, pth_resume */
 
-#define npth_yield pthread_yield
+#define npth_yield sched_yield
 
 /* Not needed.  */
 /* pth_nap */
@@ -207,6 +208,9 @@ int npth_mutex_timedlock(npth_mutex_t *mutex, const struct timespec *abstime);
 
 #define npth_mutex_unlock pthread_mutex_unlock
 
+/* Android does not have pthread_rwlock_* at all */
+#ifndef __ANDROID__
+
 #define npth_rwlockattr_t pthread_rwlockattr_t
 #define npth_rwlockattr_init pthread_rwlockattr_init
 #define npth_rwlockattr_destroy pthread_rwlockattr_destroy
@@ -248,6 +252,8 @@ int npth_rwlock_timedwrlock (npth_rwlock_t *rwlock,
 			     const struct timespec *abstime);
 #define npth_rwlock_unlock  pthread_rwlock_unlock
 
+#endif /* NOT __ANDROID */
+
 typedef pthread_cond_t npth_cond_t;
 #define NPTH_COND_INITIALIZER PTHREAD_COND_INITIALIZER
 /* For now, we don't support any cond attributes.  */
