diff --git a/npth/configure.ac b/npth/configure.ac
index c32480e..8d10564 100644
--- a/npth/configure.ac
+++ b/npth/configure.ac
@@ -215,7 +215,17 @@ if test "$have_w32_system" = no; then
   if test "$have_pthread" != no; then
     AC_DEFINE(HAVE_PTHREAD,1,[Define if we have pthread.])
     AC_CHECK_FUNCS([pthread_tryjoin_np pthread_setname_np pthread_getname_np])
-    AC_CHECK_FUNCS([pthread_mutex_timedlock])
+    # Android seems to include pthread_mutex_timedlock() in its headers, but
+    # it is not implemented.  Which means we can't really detect that.
+    # http://code.google.com/p/android/issues/detail?id=2138
+    case "${host}" in
+         *-*-android*)
+         echo "Android does not actually have pthread_mutex_timedlock()"
+         ;;
+         *)
+         AC_CHECK_FUNCS([pthread_mutex_timedlock])
+         ;;
+    esac
   fi
 fi
 
diff --git a/npth/src/npth.h.in b/npth/src/npth.h.in
index 9fdaa55..3ef2012 100644
--- a/npth/src/npth.h.in
+++ b/npth/src/npth.h.in
@@ -55,6 +55,7 @@
 #@INSERT_SYS_SELECT_H@
 
 #include <pthread.h>
+#include <sched.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -110,7 +111,7 @@ int npth_create(npth_t *thread, const npth_attr_t *attr,
 /* No can do! */
 /* pth_suspend, pth_resume */
 
-#define npth_yield pthread_yield
+#define npth_yield sched_yield
 
 /* Not needed.  */
 /* pth_nap */
